### PR TITLE
Ionic shopping tool

### PR DESCRIPTION
**Pull request by stewartjarod**
# Description

Ionic is a shopping assistant tool for llms. I've added the tool, which depends on our Ionic API SDK library.

While there, I updated the typing packages.

**Note**: _I had issues with running tests locally, even on main, so I'm not entirely sure if this is good to merge without a CI/CD step for tests._ 

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [x] New Loader/Tool
- [ ] Bug fix / Smaller change
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [x] Added new notebook (that tests end-to-end)
- [ ] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have added a library.json file if a new loader/tool was added
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

**Comment by jerryjliu** [comment]
out of curiosity @stewartjarod what were the issues you ran into in running tests?

File: pyproject.toml [hunk 1]
```diff
@@ -27,8 +27,8 @@ pytest = "7.2.1"
 pytest-dotenv = "0.5.2"
 pytest_httpserver = "1.0.8"
 pytest-mock = "3.11.1"
-typing-inspect = "0.8.0"
-typing_extensions = "^4.5.0"
+typing-inspect = "0.9.0"
```
Comment (jerryjliu) [inline pyproject.toml:30]: why this change

**Review by jerryjliu — COMMENTED** [review]
this is a cool idea! 

to make sure, the main capability the API offers is query right? not actually being able to make a purchase decision yet?

**Comment by stewartjarod** [comment]
I wasn't able to get the async tests to pass.

Right, at this time we provide a query for products and recommendations. Looking to expand from there. 😀

File: pyproject.toml (hunk 1 above)
Comment (stewartjarod) [reply pyproject.toml:30]: I have the understanding with these Python types that it is good to keep them up to date and that they are entirely backward compatible. I could be wrong! I come from mostly a TS background and try to use types for everything.

**Comment by stewartjarod** [comment]
Ah, that's a weird failure. While I'm figuring this out, @jerryjliu do you want me to revert the package update for typings?

File: pyproject.toml (hunk 1 above)
Comment (stewartjarod) [reply pyproject.toml:30]: I can roll this back if it will cause issues

**Comment by stewartjarod** [comment]
Tests should pass now 👍

**Comment by stewartjarod** [comment]
Ah, that's a new one. Type responses like that must not be supported in 3.8. Sorry for the issues here with tests. Adding future annotations seemed to do the trick here.

**Review by jerryjliu — APPROVED** [review]
